PWGLF/Tasks/LFNucleiBATask.cxx

```diff
@@ -36,50 +36,56 @@ using namespace o2::framework;
-  OutputObj<TH2F> h2KaonVspNSigmaTOF{TH2F("h2KaonVspNSigmaTOF", "NSigmaTOF(Ka) vs p; Momentum(p) in GeV; NSigmaTOF", 100, 0, 10., 200, -10, 10.)};
-  OutputObj<TH2F> h2ProtonVspNSigmaTOF{TH2F("h2ProtonVspNSigmaTOF", "NSigmaTOF(proton) vs p; Momentum(p) in GeV; NSigmaTOF", 100, 0, 10., 200, -10, 10.)};
-  OutputObj<TH2F> h2DeuteronVspNSigmaTOF{TH2F("h2DeuteronVspNSigmaTOF", "NSigmaTOF(D) vs p; Momentum(p) in GeV; NSigmaTOF", 100, 0, 10., 200, -10, 10.)};
-  OutputObj<TH2F> h2HeliumVspNSigmaTOF{TH2F("h2HeliumVspNSigmaTOF", "NSigmaTOF(He) vs p; Momentum(p) in GeV; NSigmaTOF", 100, 0, 10., 200, -10, 10.)};
-  OutputObj<TH1F> h1ProtonSpectra{TH1F("h1ProtonSpectra", "#it{p}_{T} (p); #it{p}_{T} (GeV/#it{c}); counts", 800, 0, 8)};
-  OutputObj<TH1F> h1DeuteronSpectra{TH1F("h1DeuteronSpectra", "#it{p}_{T} (d); #it{p}_{T} (GeV/#it{c}); counts", 800, 0, 8)};
-  OutputObj<TH1F> h1HeliumSpectra{TH1F("h1HeliumSpectra", "#it{p}_{T} (He); #it{p}_{T} (GeV/#it{c}); counts", 800, 0, 8)};
-
-  // trackQA
-  // OutputObj<TH1F> h1Y{TH1F("h1Y", "rapidity; Y; counts", 200, -10.0, 10)};
-  OutputObj<TH1F> h1Eta{TH1F("h1Eta", "pseudoRapidity; #eta; counts", 200, -1.0, 1.0)};
-  OutputObj<TH1F> h1VarPhi{TH1F("h1VarPhi", "#phi; #phi; counts", 80, -0.5, 7.5)};
-  OutputObj<TH2F> h2EtaVsPhi{TH2F("h2EtaVsPhi", "#eta (TOF) vs #phi; #eta; #phi", 200, -1.0, 1.0, 80, -0.5, 7.5)};
-  OutputObj<TH1F> h1pT{TH1F("h1pT", "Track #it{p}_{T}; #it{p}_{T} (GeV/#it{c}); counts", 1000, 0., 10)};
-  OutputObj<TH1F> h1p{TH1F("h1p", "Track momentum; p (GeV/#it{c}); counts", 1000, 0., 10.)};
-
-  // TPCQA
-  OutputObj<TH1F> h1TPCncr{TH1F("h1TPCncr", "number of crossed rows in TPC; TPCncr; counts", 150, 60, 170)};
-  OutputObj<TH1F> h1rTPC{TH1F("h1rTPC", "ratio of ncr over findable in TPC; rTPC; counts", 200, 0.9, 1.8)};
-  OutputObj<TH1F> h1chi2ITS{TH1F("h1chi2ITS", "#chi^{2}_{ITS}/n_{ITS}; #chi^{2}_{ITS}/n_{ITS};counts", 51, -0.5, 50.5)};
-  OutputObj<TH1F> h1chi2TPC{TH1F("h1chi2TPC", "#chi^{2}_{ITS}/n_{TPC}; #chi^{2}_{ITS}/n_{TPC}; counts", 11, -0.5, 10.5)};
-
-  // TOF mass histograms
-  OutputObj<TH2F> h2TOFmassVsPt{TH2F("h2TOFmassVsPt", "h2TOFmassVsPt; TOFmass; #it{p}_{T} (GeV)", 600, 0., 3., 500, 0., 5.)};
-
-  // TOF mass squared histograms
-  OutputObj<TH2F> h2TOFmass2ProtonVsPt{TH2F("h2TOFmass2ProtonVsPt", "#Delta M^{2} (p) vs #it{p}_{T}; #Delta M^{2} (p); #it{p}_{T} (GeV/#it{c})", 600, -3., 3., 800, 0., 8.)};
-  OutputObj<TH2F> h2TOFmass2DeuteronVsPt{TH2F("h2TOFmass2DeuteronVsPt", "#Delta M^{2} (d) vs #it{p}_{T}; #Delta M^{2} (d); #it{p}_{T} (GeV/#it{c})", 1000, -5., 5., 800, 0., 8.)};
-  OutputObj<TH2F> h2TOFmass2HeliumVsPt{TH2F("h2TOFmass2HeliumVsPt", "#Delta M^{2} (He) vs #it{p}_{T}; #Delta M^{2} (He); #it{p}_{T} (GeV/#it{c})", 1800, -9., 9., 800, 0., 8.)};
+  HistogramRegistry hevents{"HistosEvents", {}, OutputObjHandlingPolicy::AnalysisObject};
+  HistogramRegistry hqa{"HistosQA", {}, OutputObjHandlingPolicy::AnalysisObject};
```

Comment:
I don' think it is useful to have multiple registries unless you plan to have histograms with the same name, but this is also not optimal, maybe you could use only one?